pkg/manager/manager.go

```diff
@@ -478,3 +495,21 @@ func (m *manager) getUnusedVlanList(brif []netlink.Link, vlans []int) (unusedVla
 
 	return unusedVlans
 }
+
+func (l vlanlock) Lock() error {
+	dirpath := filepath.Dir(l.lock.Path())
+	if err := os.MkdirAll(dirpath, 0700); err != nil {
+		return fmt.Errorf("failed to create lock directory(%q): %v", dirpath, err)
+	}
+
+	return l.lock.Lock()
+}
+
+func (l vlanlock) Unlock() error {
+	dirpath := filepath.Dir(l.lock.Path())
```

Comment:
I think this is not required for unlock. In situation when lock is held by the process and somebody deleted the file in the middle, then unlock will still work even if file is not available, because lock is on the FD not on the file.